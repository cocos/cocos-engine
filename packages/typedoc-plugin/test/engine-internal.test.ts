import ps from 'path';
import { runTest } from './test-util';

test(`fix type arguments`, async () => {
    await runTest(ps.join(__dirname, './inputs/link/'), 'engine-internal.ts');
});
