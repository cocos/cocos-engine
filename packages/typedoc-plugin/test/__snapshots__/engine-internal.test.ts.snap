// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fix type arguments 1`] = `
Object {
  "categories": undefined,
  "ccCategories": Array [],
  "children": Array [
    Object {
      "categories": undefined,
      "children": Array [
        Object {
          "categories": undefined,
          "children": undefined,
          "comment": undefined,
          "defaultValue": undefined,
          "extendedBy": undefined,
          "extendedTypes": undefined,
          "flags": Object {},
          "getSignature": undefined,
          "groups": undefined,
          "id": 10,
          "implementationOf": undefined,
          "implementedBy": undefined,
          "implementedTypes": undefined,
          "indexSignature": undefined,
          "inheritedFrom": undefined,
          "kind": 512,
          "kindString": "Constructor",
          "name": "constructor",
          "originalName": undefined,
          "overwrites": undefined,
          "setSignature": undefined,
          "signatures": Array [
            Object {
              "comment": undefined,
              "flags": Object {},
              "id": 11,
              "implementationOf": undefined,
              "inheritedFrom": undefined,
              "kind": 16384,
              "kindString": "Constructor signature",
              "name": "new Class2",
              "originalName": undefined,
              "overwrites": undefined,
              "parameters": undefined,
              "type": Object {
                "externalUrl": undefined,
                "id": 6,
                "name": "Class2",
                "type": "reference",
                "typeArguments": undefined,
              },
              "typeParameter": undefined,
            },
          ],
          "sources": undefined,
          "type": undefined,
          "typeParameters": undefined,
        },
      ],
      "comment": undefined,
      "defaultValue": undefined,
      "extendedBy": undefined,
      "extendedTypes": undefined,
      "flags": Object {},
      "getSignature": undefined,
      "groups": Array [
        Object {
          "categories": undefined,
          "children": Array [
            10,
          ],
          "kind": 512,
          "title": "Constructors",
        },
      ],
      "id": 6,
      "implementationOf": undefined,
      "implementedBy": undefined,
      "implementedTypes": undefined,
      "indexSignature": undefined,
      "inheritedFrom": undefined,
      "kind": 128,
      "kindString": "Class",
      "name": "Class2",
      "originalName": undefined,
      "overwrites": undefined,
      "setSignature": undefined,
      "signatures": undefined,
      "sources": Array [
        Object {
          "character": 13,
          "fileName": "engine-internal.ts",
          "line": 6,
        },
      ],
      "type": undefined,
      "typeParameters": undefined,
    },
    Object {
      "categories": undefined,
      "children": Array [],
      "comment": undefined,
      "defaultValue": undefined,
      "extendedBy": undefined,
      "extendedTypes": undefined,
      "flags": Object {},
      "getSignature": undefined,
      "groups": Array [],
      "id": 27,
      "implementationOf": undefined,
      "implementedBy": undefined,
      "implementedTypes": undefined,
      "indexSignature": undefined,
      "inheritedFrom": undefined,
      "kind": 256,
      "kindString": "Interface",
      "name": "Interface2",
      "originalName": undefined,
      "overwrites": undefined,
      "setSignature": undefined,
      "signatures": undefined,
      "sources": Array [
        Object {
          "character": 17,
          "fileName": "engine-internal.ts",
          "line": 55,
        },
      ],
      "type": undefined,
      "typeParameters": undefined,
    },
    Object {
      "categories": undefined,
      "children": undefined,
      "comment": undefined,
      "defaultValue": "...",
      "extendedBy": undefined,
      "extendedTypes": undefined,
      "flags": Object {
        "isConst": true,
      },
      "getSignature": undefined,
      "groups": undefined,
      "id": 21,
      "implementationOf": undefined,
      "implementedBy": undefined,
      "implementedTypes": undefined,
      "indexSignature": undefined,
      "inheritedFrom": undefined,
      "kind": 32,
      "kindString": "Variable",
      "name": "obj2",
      "originalName": undefined,
      "overwrites": undefined,
      "setSignature": undefined,
      "signatures": undefined,
      "sources": Array [
        Object {
          "character": 13,
          "fileName": "engine-internal.ts",
          "line": 38,
        },
      ],
      "type": Object {
        "declaration": Object {
          "categories": undefined,
          "children": Array [],
          "comment": undefined,
          "defaultValue": undefined,
          "extendedBy": undefined,
          "extendedTypes": undefined,
          "flags": Object {},
          "getSignature": undefined,
          "groups": Array [],
          "id": 22,
          "implementationOf": undefined,
          "implementedBy": undefined,
          "implementedTypes": undefined,
          "indexSignature": undefined,
          "inheritedFrom": undefined,
          "kind": 65536,
          "kindString": "Type literal",
          "name": "__type",
          "originalName": undefined,
          "overwrites": undefined,
          "setSignature": undefined,
          "signatures": undefined,
          "sources": Array [
            Object {
              "character": 20,
              "fileName": "engine-internal.ts",
              "line": 38,
            },
          ],
          "type": undefined,
          "typeParameters": undefined,
        },
        "type": "reflection",
      },
      "typeParameters": undefined,
    },
  ],
  "comment": undefined,
  "flags": Object {},
  "groups": Array [
    Object {
      "categories": undefined,
      "children": Array [
        6,
      ],
      "kind": 128,
      "title": "Classes",
    },
    Object {
      "categories": undefined,
      "children": Array [
        27,
      ],
      "kind": 256,
      "title": "Interfaces",
    },
    Object {
      "categories": undefined,
      "children": Array [
        21,
      ],
      "kind": 32,
      "title": "Variables",
    },
  ],
  "id": 0,
  "kind": 1,
  "kindString": "Project",
  "name": "typedoc-plugin-cc",
  "originalName": "",
  "sources": Array [
    Object {
      "character": 0,
      "fileName": "engine-internal.ts",
      "line": 4,
    },
  ],
}
`;
